src/core_ct/core.py

```diff
@@ -94,20 +98,77 @@ def trim(self, axis: int, loc_start: int, loc_end: int | None = None) -> None:
 
         match axis:
             case 0:
-                self.pixel_array = self.pixel_array[
+                new_pixel_array = self.pixel_array[
                     loc_start : len(self.pixel_array) - loc_end
                 ]
             case 1:
-                self.pixel_array = self.pixel_array[
+                new_pixel_array = self.pixel_array[
                     :, loc_start : len(self.pixel_array[0]) - loc_end
                 ]
             case 2:
-                self.pixel_array = self.pixel_array[
+                new_pixel_array = self.pixel_array[
                     :, :, loc_start : len(self.pixel_array[0, 0]) - loc_end
                 ]
             case _:
                 raise ValueError("axis must be a value between 0 and 2 (inclusive)")
 
+        return Core(new_pixel_array, self.pixel_dimensions)
+
+    def trim_by_percent(self, axis: int, percent_on_left: float,
+                        percent_on_right: float | None = None) -> Core:
+        """
+        Reduces the dimensions of the core along a specified axis.
+
+        Get a three-dimensional slice of the core scan by trimming off a percent
+        on the requested axis. This function is symmetrical by default.
+
+        Arguments:
+        ---------
+            axis: integer either 0,1,2 specifying which dimension to collapse:
+                    0 corresponds to x-axis
+                    1 corresponds to y-axis
+                    2 corresponds to z-axis
+            percent_on_left: percent to trim off the left side.
+            percent_on_right: percent to trim off the right side.
+
+        Returns:
+        -------
+            A new trimmed core object
+
+        Raises:
+        ------
+            ValueError if axis is a value other than 0, 1, or 2
+        """
+        if percent_on_right is None:
+            percent_on_right = percent_on_left
+
+        if percent_on_left > 0.5 or percent_on_right > 0.5:
+            raise ValueError("Percents must be a less than 0.5")
+
+        match axis:
+            case 0:
```

Comment:
I think this function should just make a call to `trim` after figuring out the correct indices. This would avoid copy pasted code and make it easier to maintain

This also makes it so `trim_by_percent` doesn't have to worry about the axes at all, just pass them along to `trim`.